src/test/java/org/isf/medicalstock/test/Tests.java

```diff
@@ -1194,6 +1197,27 @@ public void testMovementGetOrigin() throws Exception {
 		assertThat(movement).isNotNull();
 		assertThat(movement.getSupplier()).isEqualTo(movement.getOrigin());
 	}
+	
+	@Test
+	public void testDeleteLastMovement() throws Exception {
+		int code = setupTestMovement(false);
+		Optional<Movement> movement = movementIoOperationRepository.findById(code); 
+		assertThat(movement).isPresent();
+		movBrowserManager.deleteLastMovement(movement.get());
+		Optional<Movement> movement2 = movementIoOperationRepository.findById(code); 
+		assertThat(movement2).isNotPresent();
+	}
+	
+	@Test
+    public void testDeleteLastMovementDenied() throws Exception {
+		int code = setupTestMovement(false);
+		Optional<Movement> movement = movementIoOperationRepository.findById(code); 
+		assertThat(movement).isPresent();
+		Movement mov = movement.get();
+		MovementType movType = new MovementType("discharge","Discharge","-");
+		mov.setType(movType);
+		assertThrows(UndeclaredThrowableException.class, () -> movBrowserManager.deleteLastMovement(mov));
+    }
```

Comment:
I don't get this test, we try to delete a movement after changing is MovementType? What this mean?
Please fix also indentation.

Reply:
we denied to delete movement if the last movement is discharge type and the medical has been used in the Ward by patient, for internal usage or tranfert to other ward. so I am try to fix the param and test

Reply:
ok, let me know when I can check again, thanks!

Reply:
@mwithi  I updated the method you can check if it is okay for you.

Reply:
I've widely tested and it works fine for me, please rename aXXX to the next availbel number and (in step_04 too) and we can merge.
Thanks a lot!